src/include/homestore/replication/repl_dev.h

```diff
@@ -170,7 +170,7 @@ class ReplDevListener {
     virtual void on_replica_stop() = 0;
 
 private:
-    ReplDev* m_repl_dev;
+    shared< ReplDev > m_repl_dev;
```

Comment:
Doesn't this add a circular dependency on ReplDev? It can never be destroyed now since it will never destroy the Listener...and so on.

Reply:
yes, it will cause circular dependency. i have rollback this change, will deal with this issue in a later PR